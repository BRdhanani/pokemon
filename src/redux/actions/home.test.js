import ApiRequestService from '../../services/api.service';
import {
  getPokemonDetail,
  getPokemonGender,
  getPokemonType,
  getPokemons,
} from './home';

jest.mock('../../services/api.service');

describe('home actions', () => {
  beforeEach(() => jest.resetAllMocks());

  it('should get PokemonDetail', () => {
    const dispatcher = getPokemonDetail('bulbasaur');

    dispatcher(jest.fn());

    expect(ApiRequestService).toMatchSnapshot();
  });

  it('should get Pokemons', () => {
    const dispatcher = getPokemons();

    dispatcher(jest.fn());

    expect(ApiRequestService).toMatchSnapshot();
  });

  it('should get PokemonType', () => {
    const dispatcher = getPokemonType();

    dispatcher(jest.fn());

    expect(ApiRequestService).toMatchSnapshot();
  });

  it('should get PokemonGender', () => {
    const dispatcher = getPokemonGender();

    dispatcher(jest.fn());

    expect(ApiRequestService).toMatchSnapshot();
  });
});
