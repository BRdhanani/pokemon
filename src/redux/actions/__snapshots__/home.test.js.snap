// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detail actions should get PokemonEgggroup 1`] = `
ApiRequestService {
  "axiosService": [MockFunction],
  "baseURL": "https://pokeapi.co/api/v2",
  "constructor": [MockFunction],
  "getApi": [MockFunction],
}
`;

exports[`detail actions should get PokemonSpecies 1`] = `
ApiRequestService {
  "axiosService": [MockFunction],
  "baseURL": "https://pokeapi.co/api/v2",
  "constructor": [MockFunction],
  "getApi": [MockFunction],
}
`;

exports[`home actions should get PokemonDetail 1`] = `
ApiRequestService {
  "axiosService": [MockFunction],
  "baseURL": "https://pokeapi.co/api/v2",
  "constructor": [MockFunction],
  "getApi": [MockFunction],
}
`;

exports[`home actions should get PokemonGender 1`] = `
ApiRequestService {
  "axiosService": [MockFunction],
  "baseURL": "https://pokeapi.co/api/v2",
  "constructor": [MockFunction],
  "getApi": [MockFunction],
}
`;

exports[`home actions should get PokemonState 1`] = `
ApiRequestService {
  "axiosService": [MockFunction],
  "baseURL": "https://pokeapi.co/api/v2",
  "constructor": [MockFunction],
  "getApi": [MockFunction],
}
`;

exports[`home actions should get PokemonType 1`] = `
ApiRequestService {
  "axiosService": [MockFunction],
  "baseURL": "https://pokeapi.co/api/v2",
  "constructor": [MockFunction],
  "getApi": [MockFunction],
}
`;

exports[`home actions should get Pokemons 1`] = `
ApiRequestService {
  "axiosService": [MockFunction],
  "baseURL": "https://pokeapi.co/api/v2",
  "constructor": [MockFunction],
  "getApi": [MockFunction],
}
`;
